lib/connection.js

```diff
@@ -186,7 +199,6 @@ Connection.prototype.startup = function (callback) {
 };
 
 Connection.prototype.errorConnecting = function (err, destroy, callback) {
```

Comment:
There are possible 3 calls to `errorConnecting()`:

1. When connecting the socket, as a handler of the `'error'` event. In this case, the error will be called before `isSocketOpen` is set to `true` (it will never be set). ✅ 
2. When connecting the socket, as a handler of `'timeout'` event. The same as before, but `destroy()` should be called, to prevent other events from firing ✅ 
3. When there was an error as part of `STARTUP` request / response. In this case, the socket will be open and the `isSocketOpen` flag set to `true`, and as [the `open()` caller received an error, it should call to `close()` it](https://github.com/datastax/nodejs-driver/blob/75be61652868d6ad726993300d58095dc9bf969f/lib/host-connection-pool.js#L199). But `socket.destroy()` should not be called, which currently is ❌

I'll fix the last use case.